modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/message/ErrorMessage.java

```diff
@@ -167,6 +167,11 @@ public Throwable error() {
 
     /** {@inheritDoc} */
     @Override public void prepareUnmarshal(MarshallingContext ctx) throws IgniteCheckedException {
-        err = ctx.unmarshal(errBytes);
+        assert err != null ^ errBytes != null;
```

Comment:
u need to expand assertion message here.

Reply:
unclear why we need exclusive OR here, can u explain ?

Reply:
honestly speaking I don't sure about this check, it was added for debug purpose... So it's better to get rid of this assertion